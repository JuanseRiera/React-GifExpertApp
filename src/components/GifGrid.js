import React from "react";
import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  let { data, loading } = useFetchGifs(category);

  const htmlToRender = () => {
    if (!loading && data && data.length > 0) {
      return (
        <div className="cardContainer">
          {data?.map((img) => {
            return <GifGridItem key={img.id} {...img} />;
          })}
        </div>
      );
    }
    if (!loading && (!data || data.length === 0)) {
      return (
        <>
          <p className="nombreCategoria animate__animated animate__headShake">
            No se encontraron elementos para la busqueda realizada.
          </p>
        </>
      );
    }
  };
  return (
    <>
      <h3 className="nombreCategoria animate__animated animate__flipInY">
        {category}
      </h3>
      {loading && (
        <p className="nombreCategoria animate__animated animate__flash">
          Cargando...
        </p>
      )}
      {htmlToRender()}
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
