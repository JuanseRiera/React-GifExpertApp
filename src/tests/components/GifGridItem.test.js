import React, { Component } from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas del componente gifgriditem", () => {
  const title = "Hola mundo";
  const url = "https://google.com";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Se debe mostrar el componente correctamente", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("Debe haber un parrofo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("La imagen debe tener como src la url y como alt el title", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("El div debe tener la clase card", () => {
    const div = wrapper.find("div");
    expect(div.prop("className")).toMatch("card");
  });
});
