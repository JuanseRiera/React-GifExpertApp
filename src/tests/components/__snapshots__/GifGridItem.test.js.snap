// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente gifgriditem Se debe mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeInLeft"
  >
    <img
      alt="Hola mundo"
      src="https://google.com"
    />
    <p>
      Hola mundo
    </p>
  </div>
</Fragment>
`;
